src/basicfuns.jl

```diff
@@ -11,15 +11,20 @@ julia> StatsFuns.xlogx(0)
 0.0
 ```
 """
-xlogx(x::Real) = x > zero(x) ? x * log(x) : zero(log(x))
+function xlogx(x)
+    result = x * log(x)
+    ifelse(iszero(x), zero(result), result)
```

Comment:
Can you update the docstring to reflect more precisely what the function does now? For complex numbers negative values are supported now. Though I'm not sure we should really allow them. Same below.

Reply:
If there is no real reason to disallow non-zero complex values, I don't see why we want to place the artificial restriction. scipy's xlogy allows complex values with negative real parts.

Reply:
Done. Note that for correctness I only have to check zero / non-zero `x` since that's the only condition here. But I added the tests so as to make explicit that we support this and so that it doesn't change inadvertently in the future.